common/route.c

```diff
@@ -0,0 +1,102 @@
+#include <assert.h>
+#include <ccan/err/err.h>
+#include <ccan/tal/str/str.h>
+#include <ccan/time/time.h>
+#include <common/dijkstra.h>
+#include <common/gossmap.h>
+#include <common/node_id.h>
+#include <common/pseudorand.h>
+#include <common/random_select.h>
+#include <common/route.h>
+#include <common/type_to_string.h>
+#include <inttypes.h>
+#include <stdio.h>
+
+bool route_can_carry_even_disabled(const struct gossmap *map,
+				   const struct gossmap_chan *c,
+				   int dir,
+				   struct amount_msat amount,
+				   void *unused)
+{
+	if (!gossmap_chan_set(c, dir))
+		return false;
+	if (!gossmap_chan_capacity(c, dir, amount))
+		return false;
+	return true;
+}
+
+/* Generally only one side gets marked disabled, but it's disabled. */
+bool route_can_carry(const struct gossmap *map,
+		       const struct gossmap_chan *c,
+		       int dir,
+		       struct amount_msat amount,
+		       void *arg)
+{
+	if (!c->half[dir].enabled || !c->half[!dir].enabled)
```

Comment:
Still not sure if this isn't too aggressive. I'm thinking in particular of my lnbig channel where the remote end refused to mark my incoming direction as active, despite it having sufficient capacity. It also goes against our previous advice of disabling channel directions if their capacity is unreasonably low and unlikely to carry any non-tiny HTLC. If we include the last ditch attempt that ignores disabled flags if no other path is available we can likely see how well this heuristic works.

I guess we'll have to see how well one heuristic compares to the other.

Reply:
Yes, my follow-on code has a "ignore disabled channels" fallback mode.